### PR TITLE
doc: pm: reorganization and new guide

### PR DESCRIPTION
**Pull request by gmarull**
This PR aims to improve the PM related docs:

- Splitted reference single doc into API and individual guides
- Removed redundant manual API function docs, added missing APIs
- Fixed a few small issues (e.g. references)
- Added a new device runtime PM guide. It contains a brief introduction, API design principles and goals,
some example sequence diagrams and usage guidelines.

Content of other parts (e.g. system) need to be updated or extended. This will be done in future PRs.

### CODE REVIEW

**Review by teburd — APPROVED** [review]
Some small wording tweaks, and I'm still confused by how an application might manage a device power state from this guide, which I think a user of Zephyr would be very interested to know, easily could've missed it though.

File: doc/guides/pm/system.rst [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+.. _pm_constraints:
+
+Power States Constraint
+=======================
+
+The power management subsystem allows different Zephyr components and
+applications to set constraints on various power states preventing the
+system from transitiioning into these states. This can be used by devices when executing
+tasks in background to avoid the system to go to a specific state where it would
```
[the remark this answers is not in the document]
Comment (gmarull) [reply doc/guides/pm/system.rst]: done. I wanted to improve the other guides in upcoming PRs (they're just a copy of what existed before).

File: doc/guides/pm/system.rst [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+.. code-block:: c
+
+   if (time_to_next_scheduled_event >= (state.min_residency_us + state.exit_latency))) {
+      return state
+   }
+
+Application
+-----------
+
+The power management policy is defined by the application by implementing the
```
[the remark this answers is not in the document]
Comment (gmarull) [reply doc/guides/pm/system.rst]: done

File: doc/guides/pm/device_runtime.rst [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+        ACTIVE -> SUSPENDED;
+        ACTIVE -> SUSPENDING [constraint=false]
+        SUSPENDING -> SUSPENDED [constraint=false];
+        SUSPENDED -> SUSPENDING [style=invis];
+        SUSPENDING -> ACTIVE [style=invis];
+    }
+
+The device runtime power management framework has been designed to minimize
+devices power consumption with minimal application work. Device drivers are
+the responsibles of indicating when they need the device to be operational and
```
[the remark this answers is not in the document]
Comment (gmarull) [reply doc/guides/pm/device_runtime.rst]: done

File: doc/guides/pm/device_runtime.rst [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+device. An application can, however, decide when to disable or enable runtime
+power management for a device. This can be useful, for example, if an
+application wants a particular device to be always active.
+
+Design principles
+*****************
+
+When runtime PM is enabled on a device, the system does no longer resume or
+suspend it during system power transitions. Instead, the device is fully
+responsible to indicate when it needs a device and when it does not. The device
```
[the remark this answers is not in the document]
Comment (gmarull) [reply doc/guides/pm/device_runtime.rst]: I have rephrased the sentence, I hope it's more clear now. What this means is that when system goes into any of the low power modes, devices with runtime PM enabled are ignored (this is what happens now). The device itself is the responsible to indicate via the get/put calls when it needs to be awake or suspended.

**Review by teburd — APPROVED** [review]

File: doc/guides/pm/system.rst [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+have higher wake latencies.
+
+.. _pm_constraints:
+
+Power States Constraint
+=======================
+
+The power management subsystem allows different Zephyr components and
+applications to set constraints on various power states preventing the
+system from transitiioning into these states. This can be used by devices when
```
Comment (ceolin) [inline doc/guides/pm/system.rst:48]: s/transitiioning/transitioning

**Review by ceolin — APPROVED** [review]
Just fix the typo please

File: doc/guides/pm/system.rst (hunk 5 above)
Comment (gmarull) [reply doc/guides/pm/system.rst]: done

**Pull request merged**